CDCgov/prime-reportstream
# DTM/DT/(TS) mapping updates

This PR updates some DTM/DT mappings to match the v2-to-fhir inventory while preserving HL7 DTM/DTs in and out.

Test Steps:
1. Run the integration tests
2. Convert a sample hl7 message containing PID.7, NK1.8/9, and OBR.22 to fhir and back and observe the results

## Changes
- Updated mappings for PID.7, PID.29, and OBR.22 from v2-to-fhir to include extension with the original HL7 string (inlined)
- Updated mappings from HL7 back to PID.7, PID.29, and OBR.22 to use new sub-schema `DTM.yml`
- Added mapping for NK1.8 and NK1.9
- Added _many_ tests for different combinations of datatypes and dateTime precision
- Adjusted _many_ existing tests

## Checklist

### Testing

## Linked Issues
- Addresses #11664
